cgroup1/utils.go

```diff
@@ -131,11 +133,19 @@ func hugePageSizes() ([]string, error) {
 }
 
 func readUint(path string) (uint64, error) {
-	v, err := os.ReadFile(path)
+	f, err := os.Open(path)
 	if err != nil {
 		return 0, err
 	}
-	return parseUint(strings.TrimSpace(string(v)), 10, 64)
+	defer f.Close()
+	b := new(bytes.Buffer)
+	b.Reset()
+	_, err = io.Copy(b, f)
+	if err != nil {
+		return 0, err
+	}
+
+	return parseUint(strings.TrimSpace(b.String()), 10, 64)
```

Comment:
Seems just using a raw byte slice cuts allocs down to 4 (although I get 7 allocs on my machine while testing your change instead of 6 as you posted). Doesn't need to be 80 bytes either, but > 64 might be necessary seeming as how this code took precautions to trim the string, so I'm assuming there's some cases where there may be trailing space in these files. 

```go
f, err := os.Open(path)
if err != nil {
	return 0, err
}
defer f.Close()

buf := make([]byte, 80)
n, err := f.Read(buf)
if err != nil {
	return 0, err
}

return parseUint(strings.TrimSpace(string(buf[:n])), 10, 64)
```

Reply:
```shell
goos: linux
goarch: amd64
pkg: github.com/containerd/cgroups/v3/cgroup1
cpu: AMD Ryzen 9 5900X 12-Core Processor            
BenchmarkReaduint64-24            849498              1342 ns/op             136 B/op          4 allocs/op
PASS
ok      github.com/containerd/cgroups/v3/cgroup1        1.160s
```

Reply:
yeah! It is significantly lower. I used buffer because of TrimSpace but adding extra bytes works as well.
I think our benchmarks are different as well (mine ends with -8 and yours with -24); maybe that is why allocs are different.

Reply:
I think we can benchmark bytebuffer in places where we are doing Readfile in entire package and see if we can reduce allocations in follow up PRs.

Reply:
That's fine with me